.github/workflows/codeql.yml

```diff
@@ -0,0 +1,77 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
```

Comment:
[q] : does it get triggered on merge to `master`? how do we know about a potential problem, will it post a comment to a pr or smth like that?

Reply:
It will post comments in PRs, and anything in `master` will get added to the `Security` tab for the repository. 😄 More info in the documentation: https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/about-code-scanning